packages/synapse-interface/utils/hooks/useStaleQuoteRefresher.ts

```diff
@@ -0,0 +1,34 @@
+import { isNull, isNumber } from 'lodash'
+import { useEffect } from 'react'
+
+import { BridgeQuote } from '@/utils/types'
+import { calculateTimeBetween } from '@/utils/time'
+import { useIntervalTimer } from '@/utils/hooks/useIntervalTimer'
+
+/**
+ * Refreshes quotes based on selected stale timeout duration.
+ * Will refresh quote when browser is active and wallet prompt is not pending.
+ */
+export const useStaleQuoteRefresher = (
+  quote: BridgeQuote,
+  refreshQuoteCallback: () => Promise<void>,
+  isQuoteLoading: boolean,
+  isWalletPending: boolean,
+  staleTimeout: number = 15000 // 15_000ms or 15s
+) => {
+  const quoteTime = quote?.timestamp
+  const isValidQuote = isNumber(quoteTime) && !isNull(quoteTime)
+  const currentTime = useIntervalTimer(staleTimeout, !isValidQuote)
+
+  useEffect(() => {
+    if (isValidQuote && !isQuoteLoading && !isWalletPending) {
+      const timeDifference = calculateTimeBetween(currentTime, quoteTime)
+      const isStaleQuote = timeDifference >= staleTimeout
+      if (isStaleQuote) {
+        document.addEventListener('mousemove', refreshQuoteCallback, {
+          once: true,
+        })
+      }
+    }
+  }, [currentTime])
```

Comment:
Do we need the other deps in here?:

```
[currentTime, isValidQuote, isQuoteLoading, isWalletPending, quoteTime, staleTimeout])
```

Reply:
`currentTime` is the only dep we would require to refresh a stale quote. Including the other deps may prematurely refresh the quote in this useEffect

Reply:
`isValidQuote, isQuoteLoading, isWalletPending`: If I'm understanding this correctly, these flags determine if an event listener should be added? And we don't want to add them bc we don't want to re-attach if they change?

`quoteTime`: Is this the timestamp which staleness is calculated? Don't we need to include that bc if `quoteTime` changes, then the staleness logic needs to re-run?

`staleTimeout`: Is this the value that determines what's considered stale? So if it changes, the staleness logic would re-run here as well?

Reply:
Correct, `isValidQuote, isQuoteLoading, isWalletPending` are used to determine if an event listener should be attached, but we don't want to react to these props because: 
- If wallet action is pending, it would refresh quote in background, but does not actually update the quote used for the transaction (prevents User confusion with output amounts)
- Listening to `isQuoteLoading` could cause for infinite re-renders 
- `isValidQuote` may change frequently if the User is testing out various inputs (token / amount combos) and could effectively re-trigger a quote fetch before interval time has passed 

Including `quoteTime` would also cause infinite re-renders since the change in `quoteTime` would cause for the quote to refresh in the background, which then updates the `quoteTime` value again. 

Good catch on the `staleTimeout`! I originally expected for `staleTimeout` to be a static value when using the hook, but that's an improper assumption to make. Updated to include `staleTimeout` in the useEffect dependencies array in https://github.com/synapsecns/sanguine/pull/2607/commits/f8fadd6fc5f2cd75378403a7841e702864e29a2c